django/contrib/admin/checks.py

```diff
@@ -987,9 +988,14 @@ def _check_actions_uniqueness(self, obj):
         """Check that every action has a unique __name__."""
         names = [name for _, name, _ in obj._get_base_actions()]
         if len(names) != len(set(names)):
+            for item, count in collections.Counter(names).items():
+                if count > 1:
+                    duplicate = str(item)
+                    break
```

Comment:
This is a classic case where you may have duplicates of multiple action names but are only warned about the first one. You resolve that and then get nagged again about the next one. It would be better to list all of the duplicated names.

It also probably makes sense to just count all of the names using `collections.Counter` and look directly for duplicates; to wit:

```python
        counts = collections.Counter(names)
        duplicates = [name for name, count in counts.items() if count > 1]
        if duplicates:
            return [checks.Error(
￼                '__name__ attributes of actions defined in %s must be '
￼                'unique. Duplicated names: %s' % (obj.__class__, ', '.join(duplicates)),
￼                obj=obj.__class__,
￼                id='admin.E130',
￼            )]
```

Reply:
Nice solution. Personally I don't find it a huge nag, I think most users will find the current method acceptable.
This said, multiple is better. I'd prefer to return a list of errors rather than a single concatenated error. Is that the method I should change it to?

Reply:
@AdamDonna Agreed.

Reply:
> I'd prefer to return a list of errors rather than a single concatenated error. Is that the method I should change it to?

Sorry - I didn't get around to replying, but yes, multiple works better.